app/src/test/java/uk/gov/onelogin/ui/welcome/WelcomeScreenViewModelTest.kt

```diff
@@ -0,0 +1,243 @@
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertNull
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.mockito.kotlin.any
+import org.mockito.kotlin.eq
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.times
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+import uk.gov.android.authentication.AuthenticationError
+import uk.gov.android.authentication.LoginSession
+import uk.gov.android.authentication.TokenResponse
+import uk.gov.android.features.FeatureFlags
+import uk.gov.android.network.online.OnlineChecker
+import uk.gov.onelogin.credentialchecker.BiometricStatus
+import uk.gov.onelogin.credentialchecker.CredentialChecker
+import uk.gov.onelogin.extensions.CoroutinesTestExtension
+import uk.gov.onelogin.extensions.InstantExecutorExtension
+import uk.gov.onelogin.login.LoginRoutes
+import uk.gov.onelogin.login.biooptin.BiometricPreference
+import uk.gov.onelogin.login.biooptin.BiometricPreferenceHandler
+import uk.gov.onelogin.login.ui.welcome.WelcomeScreenViewModel
+import uk.gov.onelogin.login.usecase.VerifyIdToken
+import uk.gov.onelogin.mainnav.nav.MainNavRoutes
+import uk.gov.onelogin.repositiories.TokenRepository
+import uk.gov.onelogin.tokens.usecases.AutoInitialiseSecureStore
+import uk.gov.onelogin.tokens.usecases.GetPersistentId
+
+@OptIn(ExperimentalCoroutinesApi::class)
+@ExtendWith(InstantExecutorExtension::class, CoroutinesTestExtension::class)
+class WelcomeScreenViewModelTest {
+    private val mockContext: Context = mock()
+    private val mockLoginSession: LoginSession = mock()
+    private val mockCredChecker: CredentialChecker = mock()
+    private val mockBioPrefHandler: BiometricPreferenceHandler = mock()
+    private val mockTokenRepository: TokenRepository = mock()
+    private val mockAutoInitialiseSecureStore: AutoInitialiseSecureStore = mock()
+    private val mockVerifyIdToken: VerifyIdToken = mock()
+    private val mockFeatureFlags: FeatureFlags = mock()
+    private val mockGetPersistentId: GetPersistentId = mock()
+    private val mockOnlineChecker: OnlineChecker = mock()
+
+    private val observer: Observer<String> = mock()
+    private val testAccessToken = "testAccessToken"
+    private var testIdToken: String? = "testIdToken"
+    private val tokenResponse = TokenResponse(
+        "testType",
+        testAccessToken,
+        1L,
+        testIdToken,
+        "testRefreshToken"
+    )
+
+    private val viewModel = WelcomeScreenViewModel(
+        mockContext,
+        mockLoginSession,
+        mockCredChecker,
+        mockBioPrefHandler,
+        mockTokenRepository,
+        mockAutoInitialiseSecureStore,
+        mockVerifyIdToken,
+        mockFeatureFlags,
+        mockGetPersistentId,
+        mockOnlineChecker
+    )
+
+    @BeforeEach
+    fun setup() {
+        viewModel.next.observeForever(observer)
```

Comment:
No `@AfterEach fun tearDown()` to remover the observer? Considering the ViewModel is a value this would mean the observer would leak after each test, no?

Reply:
fixed